src/main/java/htsjdk/variant/utils/SAMSequenceDictionaryExtractor.java

```diff
@@ -75,30 +85,53 @@ SAMSequenceDictionary extractDictionary(final File dictionary) {
                     CloserUtil.close(bufferedLineReader);
                 }
             }
+
+            @Override
+            SAMSequenceDictionary extractDictionary(Path dictionary) {
+                return extractDictionary(dictionary.toFile());
+            }
         },
         SAM(IOUtil.SAM_FILE_EXTENSION, BamFileIoUtils.BAM_FILE_EXTENSION) {
             @Override
             SAMSequenceDictionary extractDictionary(final File sam) {
                 return SamReaderFactory.makeDefault().getFileHeader(sam).getSequenceDictionary();
             }
+
+            @Override
+            SAMSequenceDictionary extractDictionary(Path sam) {
+                return SamReaderFactory.makeDefault().getFileHeader(sam).getSequenceDictionary();
+            }
         },
         VCF(IOUtil.VCF_EXTENSIONS) {
             @Override
             SAMSequenceDictionary extractDictionary(final File vcf) {
-                VCFFileReader vcfFileReader = null;
+                VCFPathReader vcfReader = null;
                 try {
-                    vcfFileReader = new VCFFileReader(vcf, false);
-                    return vcfFileReader.getFileHeader().getSequenceDictionary();
+                    vcfReader = new VCFPathReader(vcf.toPath(), false);
+                    return vcfReader.getFileHeader().getSequenceDictionary();
                 } finally {
-                    CloserUtil.close(vcfFileReader);
+                    CloserUtil.close(vcfReader);
```

Comment:
Can you replace this `close()` with try-with-resources?

Reply:
done